File: C#/CWE-287.cs
using System;
using System.Collections.Generic;

public class Authentication
{
    private Dictionary<string, string> users = new Dictionary<string, string>
    {
        { "admin", "password123" },
        { "user", "userpass" }
    };

    public bool Login(string username, string password)
    {
        // Неправильная аутентификация: проверяется только наличие имени пользователя
        if (users.ContainsKey(username))
        {
            Console.WriteLine("Authentication successful!");
            return true;
        }

        Console.WriteLine("Authentication failed!");
        return false;
    }
}

public class Program
{
    public static void Main()
    {
        Authentication auth = new Authentication();

        Console.WriteLine("Enter username:");
        string username = Console.ReadLine();

        Console.WriteLine("Enter password:");
        string password = Console.ReadLine();

        auth.Login(username, password);
    }
}

File: PHP/CWE-287.php
<?php
// A hardcoded username and password for demonstration purposes
$hardcoded_username = "admin";
$hardcoded_password = "password123";

// Retrieving user input from a form (in a real scenario, this would come from POST data)
$input_username = $_POST['username'];
$input_password = $_POST['password'];

// Improper authentication check
if ($input_username == $hardcoded_username && $input_password == $hardcoded_password) {
    echo "Authentication successful! Welcome, " . htmlspecialchars($input_username) . "!";
} else {
    echo "Authentication failed. Invalid username or password.";
}
?>
